wordpress-mobile/WordPress-iOS
# Stats Period Details: Optimize table view updates by integrating diffable data source

Fixes #22720
Related PRs:
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22542
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22592

### Description

Introduce diffable data source into Stats Details (Traffic -> Details, Days/../Years -> Details) views. I only found 2 ViewControllers used for Stats Details:
- `PostStatsTableViewController` to show Stats for a Post
- `SiteStatsDetailTableViewController` for all the other Details views

### Solution

1. Change `ImmuTableViewHandler` to `ImmuTableDiffableViewHandler`
2. Call `diffableDataSource.apply` instead of setting `tableViewModel`
3. Build `tableViewSnapshot` instead of `tableViewModel`
4. Make required models `Hashable`
5. Added helper methods to handle a common case of building a single-section table view snapshot

### To test

Test with both `Stats Traffic` feature flag enabled and disabled:

1. Select a high traffic site
6. Open `Traffic` or `Days/../Years` tab
7. Go through each of the cards and tap "View More"
8. For Post & Pages cards click on a specific post
9. Confirm that views open and scroll without any crashes or glitches

## Regression Notes
1. Potential unintended areas of impact

Breaking Stats Details views

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:


Testing checklist:
